src/Symfony/Component/Uid/AbstractUid.php

```diff
@@ -159,6 +159,11 @@ public function equals(mixed $other): bool
         return $this->uid === $other->uid;
     }
 
+    public function hash(): mixed
```

Comment:
```suggestion
    public function hash(): string
```

Reply:
technically that wouldn't be compatible with `Ds\Hashable`, which doesn't specify a return type at all, but I suppose it can't hurt anything here.. just a question while I'm at it though, I'm not sure what the local convention is for suggestions like this, do I just accept them using the Github UI? I usually tend to redo them manually and push them in a single commit to avoid cluttering the commit history, but I'm equally happy to do either depending on what you guys prefer..

Reply:
squashing is fine
providing more specialized return types is also legal and improves type descriptions so we should do it

Reply:
You can just amend your commit locally and then force push to your branch. Pushing several commits (but no merge commits) is fine as well, we will then squash them into a single commit when merging.

Reply:
@nicolas-grekas yeah generally providing more specialized types is good, but I thought that since the purpose here is to _polyfill_ something, the idea is to get as close to the real thing as possible.. but sure, with the addition of the stricter `hash(): string` declaration to the extended variant of the interface as well as marking both variants `@internal` it totally makes sense to do it like this, thanks!

Reply:
@nicolas-grekas @xabbuh okay, I amended & force-pushed all the suggestions now if I'm not mistaken, there's just an issue with a Psalm test which doesn't like that the argument of `AbstractUid::equals()` is called `$other` and not `$obj` like in `Ds\Hashable` and I'm not sure what to do about this, sice renaming it could probably break tests for people who extend `AbstractUid` with their own subclasses..?